### PR TITLE
Support multiple CSV files in series

### PR DESCRIPTION
**Pull request by cottsay**
Since the CSVSeries supports multiple data points anyway, allow a wildcard pattern to match multiple files and add them all to the same series.

### How to test
Specify a pattern to a CSV file which matches multiple files.
Note that a test was added to verify this.

### Checklist

- [x] Git commits follow [best practices](https://chris.beams.io/posts/git-commit/) 
- [ ] Build passes in Jenkins 
- [x] Appropriate tests or explanation to why this change has no tests 
- [ ] JIRA issue is well described (problem explanation, steps to reproduce, screenshots) 
- [ ] For dependency updates: links to external changelogs and, if possible, full diffs

### CODE REVIEW

File: src/main/java/hudson/plugins/plot/CSVSeries.java [hunk 1]
```diff
@@ -109,36 +109,55 @@ public boolean getDisplayTableFlag() {
...
+            return null;
+        }
+
+        if (ArrayUtils.isEmpty(seriesFiles)) {
+            LOGGER.info("No plot data file found: " + workspaceRootDir.getName()
+                    + " " + getFile());
+            return null;
+        }
+
+        for (int index = 0; index < seriesFiles.length; index++) {
```
Comment (vgaidarji) [inline src/main/java/hudson/plugins/plot/CSVSeries.java]: how about `foreach` syntax instead?
```suggestion
        for (FilePath seriesFile : seriesFiles) {
            List<PlotPoint> seriesList = loadSeriesFile(seriesFile, buildNumber, logger);
```

File: src/test/java/hudson/plugins/plot/CSVSeriesTest.java [hunk 2]
```diff
@@ -154,4 +154,20 @@ private int getNumColumns(FilePath workspaceRootDir, String file) throws IOExcep
...
+        for (int index = 0; index < FILES.length; index++) {
+            // Create a new CSV series.
+            CSVSeries series = new CSVSeries("*.csv", "http://localhost:8080/%name%/%index%/", "INCLUDE_BY_STRING", "Avg", false);
+
+            LOGGER.info("Created series " + series.toString());
+
+            // load the series.
+            List<PlotPoint> points = series.loadSeries(workspaceRootDir, 0, System.out);
+            LOGGER.info("Got " + points.size() + " plot points");
+            testPlotPoints(points, 2);  // Avg is found in 2 files, each containing 1 line
```
[the remark this answers is not in the document]
Comment (cottsay) [reply src/test/java/hudson/plugins/plot/CSVSeriesTest.java]: I created some separate test collateral for this case, and used the existing tests to verify functionality. Should be a lot less fragile.

**Comment by cottsay** [comment]
Thanks very much for the swift and helpful feedback, @vgaidarji. I believe I have addressed all of your concerns.

Please squash these commits prior to merging - I left them fanned out to facilitate review.

**Comment by vgaidarji** [comment]
@cottsay thank you. I'll merge and release the new version soon 👍

**Comment by vgaidarji** [comment]
The new version of the plugin [2.1.3](https://github.com/jenkinsci/plot-plugin/releases/tag/plot-2.1.3) was released and should be available in [update center](https://updates.jenkins-ci.org/download/plugins/plot/) soon. But it was already uploaded to repo https://repo.jenkins-ci.org/releases/org/jenkins-ci/plugins/plot/2.1.3/.
I'll need to migrate release notes from old wiki format to GitHub and will include release notes for 2.1.3 there.

@cottsay, thank you very much for your contribution 🤝🎉

**Pull request merged**